homework_2/turtle.js

```diff
@@ -0,0 +1,214 @@
+/* [Homework] Turtle Graphics - 2 */
+
+// Create a turtle class
+class Turtle {
+  constructor(x, y) {
+    this.x = x;
+    this.y = y;
+    this.direction = 'east';
+    this.points = [[this.x, this.y]];
+    this.dir = ['east'];
+    this.paths = [];
+  }
+
+  // Set a forward method on the turtle instance
+  forward(num) {
+    if (this.direction === 'east') {
+      this.x += num;
+    } else if (this.direction === 'west') {
+      this.x -= num;
+    } else if (this.direction === 'north') {
+      this.y -= num;
+    } else {
+      this.y += num;
+    }
+    this.points.push([this.x, this.y]);
+    this.dir.push(this.direction);
+    return this;
+  }
+
+  // Set a right method on the turtle instance
+  right() {
+    const dir = ['north', 'east', 'south', 'west'];
+    let idx = dir.indexOf(this.direction);
+    idx++;
+    if (idx === undefined || idx >= dir.length) {
+      idx = 0;
+    }
+    this.direction = dir[idx];
+    return this;
+  }
+
+  // Set a left method on the turtle instance
+  left() {
+    const dir = ['north', 'east', 'south', 'west'];
+    let idx = dir.indexOf(this.direction);
+    idx--;
+    if (idx < 0) {
+      idx = dir.length - 1;
+    }
+    this.direction = dir[idx];
+    return this;
+  }
+
+  allPoints() {
+    // Merge both the designated points and the path points arrays by first creating an empty array then pushing into them
+    const finalTemp = [];
+    this.points.forEach((el) => {
+      finalTemp.push(el);
+    });
+    this.pathPoints().forEach((el) => {
+      finalTemp.push(el);
+    });
+
+    // Then create a seperate array, create a set and set them as strings.  Then remove them through the Javascript Set method and return the array with only unique values
+    let stringArray = finalTemp.map(JSON.stringify);
+    let uniqueStringArray = new Set(stringArray);
+    let uniqueArray = Array.from(uniqueStringArray, JSON.parse);
+    return uniqueArray;
+  }
+
+  print() {
+    // Set a default grid size, output variable to an empty string and a check to false
+    let gridSize = 10;
```

Comment:
`const`

Reply:
rather than having the grid size pre-defined, it should adjust itself dynamically depending on how it draws.